tests/integration/targets/azure_rm_search/tasks/main.yml

```diff
@@ -0,0 +1,172 @@
+- name: Create Azure Search name
+  set_fact:
+    search_name: "search{{ resource_group | hash('md5') | truncate(16, True, '') }}"
+
+- name: Create invalid Azure Search - Hosting Name
+  azure_rm_search:
+    name: "{{ search_name }}"
+    resource_group: "{{ resource_group }}"
+    hosting_mode: highDensity
+  register: invalid_hosting_name
+  ignore_errors: yes
+
+- name: Assert task failed
+  assert:
+    that: 
+      - invalid_hosting_name.failed == True
+
+- name: Create invalid Azure Search - Partition Count High Density
+  azure_rm_search:
+    name: "{{ search_name }}"
+    resource_group: "{{ resource_group }}"
+    sku: standard3
+    hosting_mode: highDensity
+    partition_count: 4
+  register: invalid_partition_count
+  ignore_errors: yes
+
+- name: Assert task failed
+  assert:
+    that: 
+      - invalid_partition_count.failed == True
+
+- name: Create invalid Azure Search - Partition Count
+  azure_rm_search:
+    name: "{{ search_name }}"
+    resource_group: "{{ resource_group }}"
+    partition_count: 7
+  register: invalid_partition_count
+  ignore_errors: yes
+
+- name: Assert task failed
+  assert:
+    that: 
+      - invalid_partition_count.failed == True
+
+- name: Create invalid Azure Search - Replica Count
+  azure_rm_search:
+    name: "{{ search_name }}"
+    resource_group: "{{ resource_group }}"
+    sku: standard
+    replica_count: 13
+  register: invalid_replica_count
+  ignore_errors: yes
+
+- name: Assert task failed
+  assert:
+    that: 
+      - invalid_replica_count.failed == True
+
+- name: Create invalid Azure Search - Replica Count SKU basic
+  azure_rm_search:
+    name: "{{ search_name }}"
+    resource_group: "{{ resource_group }}"
+    replica_count: 4
+    sku: basic
+  register: invalid_replica_count
+  ignore_errors: yes
+
+- name: Assert task failed
+  assert:
+    that: 
+      - invalid_replica_count.failed == True
+
+- name: Create basic Azure Search
+  azure_rm_search:
```

Comment:
Should this place use the azure_rm_search_info module?

Reply:
I've added extra verification to test info module. Thanks for your comments :)